Day9/puzzle.py

```diff
@@ -0,0 +1,33 @@
+from collections import deque
+
+def in_preamble(n, preamble):
+    for i in range(len(preamble)-1):
+        for j in range(i+1, len(preamble)):
+            if n == preamble[i] + preamble[j]:
+                return True
+    return False
+
+def find_first_violation(seq, preamble_size):
+    preamble = deque(seq[:preamble_size])
+    for i in range(preamble_size, len(seq)):
+        if in_preamble(seq[i], preamble):
+            preamble.append(seq[i])
+            preamble.popleft()
+        else: return seq[i]
+    
+def find_contiguous_set(seq, key):
+    for i in range(len(seq)):
+        for delta_i in range(len(seq)-i):
+            if sum(seq[i:i+delta_i]) == key:
+                return min(seq[i:i+delta_i]) + max(seq[i:i+delta_i])
```

Comment:
How does this avoid grabbing the number when i == index of the key and delta_i = 1? Shouldn't it be `delta_i in range(2, len(seq)-i)`

Reply:
I think you luck out in that the i for the set occurs sooner than the key

Reply:
Good point, thanks!